samples/matter/lock/src/zcl_callbacks.cpp

```diff
@@ -23,6 +23,16 @@ void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath &a
 	VerifyOrReturn(attributePath.mClusterId == DoorLock::Id &&
 		       attributePath.mAttributeId == DoorLock::Attributes::LockState::Id);
 
+	app::DataModel::Nullable<DlLockState> currentLockState;
+	Attributes::LockState::Get(GetAppTask().kLockEndpointId, currentLockState);
+
+	/* Return if current lock state is the same as given */
+	if (!currentLockState.IsNull()) {
+		if (*value == to_underlying(currentLockState.Value())) {
```

Comment:
Can it not be true given that it's the **post**-attribute-change callback? Maybe the right solution should be not handling the `AppEvent::Lock` event if `LockManager` says it's already locked?

Reply:
Oh right. Do you mean posting, but not handling it or not posting it? I believe it would be better to not post it, as otherwise on every change we will unnecessarily post one event that will be dropped every time.

Reply:
Yes, exactly.

Reply:
Ok, done, please re-check.